packages/feedback/src/screenshot/components/ScreenshotEditorv2.tsx

```diff
@@ -0,0 +1,356 @@
+
+        imageBuffer.width = imageSource.videoWidth;
+        imageBuffer.height = imageSource.videoHeight;
+      }, []),
+      onAfterScreenshot: hooks.useCallback(() => {
+        (dialog.el as HTMLElement).style.display = 'block';
+        setdisplayEditor(true);
+      }, []),
+      onError: hooks.useCallback(error => {
+        (dialog.el as HTMLElement).style.display = 'block';
+        setdisplayEditor(true);
+        onError(error);
+      }, []),
+    });
+
+    const onDraw = (e: MouseEvent): void => {
+      const graywashCanvas = graywashRef.current;
+      if (!action || !graywashCanvas) {
+        return;
+      }
+
+      const boundingRect = graywashCanvas.getBoundingClientRect();
+
+      const startX = e.clientX - boundingRect.left;
+      const startY = e.clientY - boundingRect.top;
+
+      const handleMouseMove = (e: MouseEvent): void => {
+        const endX = e.clientX - boundingRect.left;
+        const endY = e.clientY - boundingRect.top;
+
+        const rect = constructRect({ action, startX, startY, endX, endY });
+
+        // prevent drawing rect when clicking on the canvas (ie clicking delete)
+        if (action && startX != endX && startY != endY) {
+          setCurrentRect(rect);
+        }
+      };
+
+      const handleMouseUp = (e: MouseEvent): void => {
+        const endX = Math.max(0, Math.min(e.clientX - boundingRect.left, graywashCanvas.width / DPI));
+        const endY = Math.max(0, Math.min(e.clientY - boundingRect.top, graywashCanvas.height / DPI));
+        // prevent drawing rect when clicking on the canvas (ie. clicking delete)
+        if (startX != endX && startY != endY) {
+          // scale to image buffer
+          const scale = imageBuffer.width / graywashCanvas.clientWidth;
+          const rect = constructRect({
+            action,
+            startX: startX * scale,
+            startY: startY * scale,
+            endX: endX * scale,
+            endY: endY * scale,
+          });
+          setDrawCommands(prev => [...prev, rect]);
+        }
+
+        DOCUMENT.removeEventListener('mousemove', handleMouseMove);
+        DOCUMENT.removeEventListener('mouseup', handleMouseUp);
+      };
+
+      DOCUMENT.addEventListener('mousemove', handleMouseMove);
+      DOCUMENT.addEventListener('mouseup', handleMouseUp);
+    };
+
+    const handleDeleteRect = (index: number): void => {
+      const updatedRects = [...drawCommands];
+      updatedRects.splice(index, 1);
+      setDrawCommands(updatedRects);
+    };
+
+    return (
+      <div class="editor">
+        <style nonce={options.styleNonce} dangerouslySetInnerHTML={styles} />
+        <div class="editor__image-container">
+          <div class="editor__canvas-container" ref={measurementRef}>
+            <canvas ref={screenshotRef}></canvas>
+            <canvas class="editor__canvas-annotate" ref={graywashRef} onMouseDown={onDraw}></canvas>
+            <div class="editor__rect-container" ref={rectDivRef} onMouseDown={onDraw}>
```

Comment:
```suggestion
            <canvas class="editor__canvas-annotate" ref={graywashRef} onMouseDown={handleMouseDown}></canvas>
            <div class="editor__rect-container" ref={rectDivRef} onMouseDown={handleMouseDown}>
```

consistent event callback names 2/3